### PR TITLE
Deploy kind clusters in parallel

### PR DESCRIPTION
**Pull request by skitt**
This greatly reduces e2e deployment time.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Comment by skitt** [comment]
Parallelising subsequent steps involves handling multiple K8s contexts in parallel, which isn’t quite as straightforward.

**Comment by skitt** [comment]
> This greatly reduces e2e deployment time.

OK, only 30s less on the Travis job, but on systems with the images already available the change is quite significant.

File: scripts/kind-e2e/e2e.sh [hunk 1]
```diff
@@ -23,9 +28,21 @@ function kind_clusters() {
 
             sed -i -- "s/server: .*/server: https:\/\/$master_ip:6443/g" $(kind get kubeconfig-path --name="cluster$i")
             cp -r $(kind get kubeconfig-path --name="cluster$i") ${PRJ_ROOT}/output/kind-config/dapper/kind-config-cluster${i}
-            export KUBECONFIG=$(kind get kubeconfig-path --name=cluster1):$(kind get kubeconfig-path --name=cluster2):$(kind get kubeconfig-path --name=cluster3)
+            ) > ${logs[$i]} 2>&1 &
```
Comment (mangelajo) [inline scripts/kind-e2e/e2e.sh]: if we use tee here, we can see the output, although a bit mixed up..

Comment (mangelajo) [reply scripts/kind-e2e/e2e.sh]: I suggest this, which gives you feedback:

diff --git a/scripts/kind-e2e/e2e.sh b/scripts/kind-e2e/e2e.sh
index 0793d23..85d11c7 100755
--- a/scripts/kind-e2e/e2e.sh
+++ b/scripts/kind-e2e/e2e.sh
@@ -34,12 +34,15 @@ function kind_clusters() {
     done
     for i in 1 2 3; do
         if [[ pids[$i] -gt -1 ]]; then
+            tail -f ${logs[*]} &
+            TAIL_PID=$!
             wait ${pids[$i]}
             if [[ $? -ne 0 && $? -ne 127 ]]; then
                 echo Cluster $i creation failed:
                 cat $logs[$i]
             fi
             rm -f $logs[$i]
+            kill $TAIL_PID
         fi
     done
     export KUBECONFIG=$(kind get kubeconfig-path --name=cluster1):$(kind get kubeconfig-path --name=cluster2):$(kind get kubeconfig-path --name=cluster3)

Comment (mangelajo) [reply scripts/kind-e2e/e2e.sh]: The feedback will be in this form: 

Creating cluster1, logging to /tmp/tmp.77aX7hkugz...
Creating cluster2, logging to /tmp/tmp.IjrhkE6nbW...
Creating cluster3, logging to /tmp/tmp.xeuLm0TvrT...
==> /tmp/tmp.77aX7hkugz <==
Creating cluster "cluster1" ...
 • Ensuring node image (kindest/node:v1.14.2) 🖼  ...
 ✓ Ensuring node image (kindest/node:v1.14.2) 🖼
 • Preparing nodes 📦  ...

==> /tmp/tmp.IjrhkE6nbW <==
Creating cluster "cluster2" ...
 • Ensuring node image (kindest/node:v1.14.2) 🖼  ...

==> /tmp/tmp.xeuLm0TvrT <==

==> /tmp/tmp.IjrhkE6nbW <==
 ✓ Ensuring node image (kindest/node:v1.14.2) 🖼
 • Preparing nodes 📦📦📦  ...

==> /tmp/tmp.xeuLm0TvrT <==
Creating cluster "cluster3" ...
 • Ensuring node image (kindest/node:v1.14.2) 🖼  ...
 ✓ Ensuring node image (kindest/node:v1.14.2) 🖼
 • Preparing nodes 📦📦📦  ...

==> /tmp/tmp.77aX7hkugz <==
 ✓ Preparing nodes 📦
 • Creating kubeadm config 📜  ...
 ✓ Creating kubeadm config 📜
 • Starting control-plane 🕹️  ...

Comment (mangelajo) [reply scripts/kind-e2e/e2e.sh]: oops tail needs to be out of the for loop, but if I do that.. then it keeps waiting on tail even if I kill it...

**Pull request merged**